service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcRequestBuilder.java

```diff
@@ -0,0 +1,139 @@
+                    .setValue(new BooleanType(true))))
+            .addParameter(new Parameters.ParametersParameterComponent()
+                .setName("includeMedication"))
+            .addParameter(new Parameters.ParametersParameterComponent()
+                .setName("includeConsultations")
+                .addPart(new Parameters.ParametersParameterComponent()
+                    .setName("includeNumberOfMostRecent")
+                    .setValue(new IntegerType(NUMBER_OF_RECENT_CONSULTANTS))))
+            .addParameter(new Parameters.ParametersParameterComponent()
+                .setName("includeProblems"))
+            .addParameter(new Parameters.ParametersParameterComponent()
+                .setName("includeImmunisations"))
+            .addParameter(new Parameters.ParametersParameterComponent()
+                .setName("includeUncategorisedData"))
+            .addParameter(new Parameters.ParametersParameterComponent()
+                .setName("includeInvestigations"))
+            .addParameter(new Parameters.ParametersParameterComponent()
+                .setName("includeReferrals"));
+    }
+
+    public WebClient.RequestHeadersSpec<?> buildGetStructuredRecordRequest(Parameters requestBodyParameters,
+        EhrExtractStatus ehrExtractStatus) throws SSLException {
+        SslContext sslContext = buildSSLContext();
+        HttpClient httpClient = HttpClient.create().secure(t -> t.sslContext(sslContext));
+        WebClient client = buildWebClient(httpClient);
+
+        WebClient.RequestBodySpec uri = client
+            .method(HttpMethod.POST)
+            .uri(gpcConfiguration.getEndpoint());
+
+        var requestBody = fhirParser.encodeResourceToString(requestBodyParameters);
+        BodyInserter<Object, ReactiveHttpOutputMessage> bodyInserter
+            = BodyInserters.fromValue(requestBody);
+
+        var request = ehrExtractStatus.getEhrRequest();
+
+        return uri
+            .body(bodyInserter)
+            .accept(MediaType.valueOf(FHIR_CONTENT_TYPE))
+            .header(SSP_FROM, request.getFromAsid())
+            .header(SSP_TO, request.getToAsid())
+            .header(SSP_INTERACTION_ID, "urn:nhs:names:services:gpconnect:fhir:operation:gpc.getstructuredrecord-1")
+            .header(SSP_TRACE_ID, ehrExtractStatus.getConversationId())
+            .header(AUTHORIZATION, AUTHORIZATION_BEARER + gpcTokenBuilder.buildToken(ehrExtractStatus.getEhrRequest().getFromOdsCode()))
+            .header(TARGET_HOST, gpcConfiguration.getHost())
+            .header(CONTENT_LEN, valueOf(requestBody.length()))
+            .header(CONTENT_TYPE, FHIR_CONTENT_TYPE);
+    }
+
+    private SslContext buildSSLContext() throws SSLException {
+        return SslContextBuilder
+            .forClient()
+            .trustManager(InsecureTrustManagerFactory.INSTANCE)
+            .build();
+    }
+
+    private org.springframework.web.reactive.function.client.WebClient buildWebClient(HttpClient httpClient) {
```

Comment:
Is there a problem with an import? Does a WebClient class really need to be used with each package?